clusterspec/clusterspec.go

```diff
@@ -0,0 +1,59 @@
+package clusterspec
+
+import (
+	"net/url"
+	"strings"
+
+	v1alpha1core "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	v1alpha1provider "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/microerror"
+)
+
+// Factory type acts as a state holder for supplemental v1alphaprovider.Cluster
+// info.
+type Factory struct {
+	baseCluster *v1alpha1provider.Cluster
+}
+
+// NewFactory constructs new Factory
+func NewFactory(baseCluster *v1alpha1provider.Cluster) (*Factory, error) {
+	// TODO: Validate relevant Cluster fields to not be empty
+	return &Factory{baseCluster}, nil
+}
+
+// New constructs an instance of provider/v1alpha1.Cluster filled with
+// configured information.
+func (f *Factory) New(clusterGuestConfig v1alpha1core.ClusterGuestConfig) (*v1alpha1provider.Cluster, error) {
```

Comment:
Why this package is named `clusterspec` when it produces `Cluster`s? Is there some crucial part that isn't visible yet?

Reply:
It's my thinking that in this context (`cluster-operator` that creates/orders clusters), `cluster` would give wrong impression for contents of this package. The sole purpose of this package is to act as configurable factory that produces "specification" for more or less abstract Kubernetes cluster. This does not create the cluster or cover all the parts needed to create a cluster so within our rules for other structuring and naming, I think `cluster` would be misleading here. I understand your concern though, that's something I also initially thought, but then settled with this. WDYT?

Reply:
`cluster` isn't misleading for me, but I'm fine with the current name when you like it more.